site/Index.jsx

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { HashRouter as Router } from 'react-router-dom';
+import { Provider } from 'xfc';
+
+import App from 'terra-site/lib/app/App';
+import routeConfiguration from 'terra-site/lib/app/configureApp';
+
+import componentConfig from './generatedComponentConfig';
+import siteConfig from './siteConfig';
+
+const { routeConfig, navigation } = routeConfiguration(siteConfig, componentConfig);
```

Comment:
The routeConfiguration function takes the siteConfig and componentConfig and generates:

1.  `navigation`: the navigation object used to create the [`ApplicationHeader`](https://github.com/cerner/terra-site/blob/dynamic-site/src/app/ApplicationHeader.jsx) which in turn creates [`NavTabs`](https://github.com/cerner/terra-site/blob/dynamic-site/src/app/common/nav-tabs/NavTabs.jsx). Additionally, the navigation object is used to set the terra-navigation-layout's [`indexPath`](https://github.com/cerner/terra-framework/blob/daf313cf804938e6135f70be3f38f4b123b1d4d4/packages/terra-navigation-layout/src/NavigationLayout.jsx#L65) prop.

2. `routeConfig`: the configuration that is passed to the terra-navigations-layout's [`config`](https://github.com/cerner/terra-framework/blob/daf313cf804938e6135f70be3f38f4b123b1d4d4/packages/terra-navigation-layout/src/NavigationLayout.jsx#L60)  prop.
     - The routeConfig contains two objects: the content and menu configuration
          - `content`: are the terra-site's [`Components`](https://github.com/cerner/terra-site/blob/dynamic-site/src/app/components/Components.jsx) components- each which creates the react-router routes for the  component to be rendered as specified by the generatedComponentConfig file.
        - `menu`: are the terra-site's [`ComponentsMenu`](https://github.com/cerner/terra-site/blob/dynamic-site/src/app/components/ComponentsMenu.jsx) components- each create the side-menu base and sub navigation as new routes are rendered. Again these are created as specified by the generatedComponentConfig file.